src/engine/MaxConstraint.cpp

```diff
@@ -512,11 +512,10 @@ List<PiecewiseLinearConstraint::Fix> MaxConstraint::getSmartFixes( ITableau * )
 
 List<PhaseStatus> MaxConstraint::getAllCases() const
 {
-    if ( _phaseStatus != PHASE_NOT_FIXED )
+    if ( phaseFixed() )
```

Comment:
This is a quick fix to the issue (https://github.com/NeuralNetworkVerification/Marabou/issues/504) resulting from the refactored MaxConstraint. For now this is probably enough for the current master to be sound, because most of the refactored functionality is not in use. A more thorough change in the MaxConstraint is needed for the context dependent stuff to be actually usable.